pegasus/helpers/section_api_helpers.rb

```diff
@@ -208,12 +213,12 @@ def self.valid_courses(user_id = nil)
           name = ScriptConstants.teacher_dashboard_name(course[:name])
           first_category = ScriptConstants.categories(course[:name])[0] || 'other'
           position = ScriptConstants.position_in_category(name, first_category)
-          name = I18n.t("#{name}_name", default: name)
+          name = I18n.t("#{name}_name", default: name, locale: course_locale)
           name += " *" if course[:hidden]
```

Comment:
I notice that i18n is broken for hidden scripts (this is not a big deal or, currently, a deal at all since hidden scripts are only visible to admins)